### DISCUSSION TITLE
Welke data willen we metadateren?

### DISCUSSION THREAD

**Discussion by rubenvdlinde** [OpenCatalogi/.github#10] [Q&A] [answered]
Onder #7 #85 #85 #67 #75 en #67 wordt er gevraagd naar het metadateren van gevens. Die roept de vraag op welke metedata willen we over componenten, modelen, processen en documentatie opslaan?

We kunnen hiervoor naar een aantal voorbeelden en bronnen kijken

- joinup.eu
- [publiccode.yaml](https://yml.publiccode.tools/), heeft ook een aardige [editor](https://publiccode-editor.developers.italia.it/).
- [developer.overheid](https://developer.overheid.nl/) =[API Specs](https://developer.overheid.nl/api/repositories)
- [commonground componenten catalogus](https://componentencatalogus.commonground.nl/) = [API Specs](https://componentencatalogus.commonground.nl/api/components/)
- [data platform](https://stichtsevecht.dataplatform.nl/#/data)
- [data.overheid.nl]() -> [API Specs](https://docs.ckan.org/en/latest/api/index.html)
- [sdg invoervoorziening](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/maykinmedia/sdg-invoervoorziening/master/src/openapi.yaml)

In een bredere context raakt dit wellicht ook aan de wet openbaarheid bestuur

Voor het datamodel zullen we in eerste instantie te kijken naar wat er nu al gebruikt wordt om componenten en applicaties  te omschrijven en daarop uit te breiden indien nodig aan de hand extra eisen (UPL namen bijvoorbeeld).  Start is dan even de huidige componenten catalogus. Die heeft een relatief simpele opbouw.

```json
{
        "id": 19,
        "name": "Adresservice",
        "layerType": "services",
        "description": "Dit component verrijkt huisnummer en postcodes bevragingen met BAG gegevens.",
        "organisationName": "Conduction",
        "contact": "[ruben@conduction.nl](mailto:ruben@conduction.nl)",
        "repositoryUrl": "https://github.com/ConductionNL/adresservice",
        "reuseType": "zelf installeren",
        "expectedQuarter": null,
        "expectedYear": null,
        "developedBy": null,
        "status": "bruikbaar",
        "owner": {
            "id": 5,
            "fullName": "Ruben van der Linde",
            "email": "[ruben@conduction.nl](mailto:ruben@conduction.nl)",
            "pictureUrl": "https://account.pleio.nl/media/avatars/2019/07/04/60221507/large_jjMfUkz.jpg"
        },
        "createdAt": "2020-02-18T10:37:28.710861Z"
    }
```

Kijken naar voorliggende user stories is dat te weinig informatie.  Een meer waardevolle standaard lijkt [publiccode](https://yml.publiccode.tools/schema.core.html) te zijn.
Hierin zien we meteen de kernaspecten terug die we willen, algemeen beschrijving, meertaligheid en afhankelijkheden.  Het lijkt ook relatief goed uit te lijnen met het huidige datamodel van de [commonground](https://componentencatalogus.commonground.nl/api/components/).  Zo zijn bijvoorbeeld [developmentstatus](https://yml.publiccode.tools/schema.core.html#key-developmentstatus) van publiccode en status van de componenten catalogus vergelijkbaar.

Voorstel is dan ook om publiccode van een yaml naar api om te slaan, en uit te breiden met de twee properties uit de componenten catalogus die we dan missen (layerType en installationType) vanuit de huidige componenten catalogus.  De publiccode standaard ondersteund [landspecifieke afwijkingen](https://yml.publiccode.tools/country.html).  Dus daar kunnen we ze onder opnemen.

Dan eindigen we met [deze specs](https://conduction.stoplight.io/docs/publiccode) wat ons als voorbeeld opleverd:

```json
{
  "id": "497f6eca-6276-4993-bfeb-53cbbbba6f08",
  "name": "Medusa",
  "applicationSuite": {
    "id": "string",
    "name": "string"
  },
  "url": "https://example.com/italia/medusa.git",
  "landingURL": "https://example.com/italia/medusa",
  "isBasedOn": "https://github.com/italia/otello.git",
  "softwareVersion": "1.0",
  "logo": "https://avatars0.githubusercontent.com/u/34739001?s=280&v=4",
  "releaseDate": "2022-01-01",
  "platforms": "web",
  "categories": [
    "Acounting"
  ],
  "usedBy": [
    "City of Amsterdam"
  ],
  "roadmap": "https://vng.nl/agenda",
  "developmentStatus": "concept",
  "softwareType": "standalone/web",
  "layerType": "interface",
  "installationType": "self",
  "intendedAudience": {
    "countries": [
      "nl"
    ],
    "unsupportedCountries": [
      "de"
    ],
    "scope": [
      "agriculture",
      "culture"
    ]
  },
  "description": {
    "NL": {}
  }
}
```

*Voor componenten zijn we dan redelijk los, we missen echter nog wat details*

Vanuit public code zit een component vast aan één applicatie, vanuit commonground is dat eigenlijk omgekeerd en bestaad één applicatie uit meerdere componenten waarbij één component in meerdere applicaties kan voorkomen. 

Dat pleit er voor om van applicationSuite een object te maken ipv een sting (reeds verwerkt in voorbeeld api), dat object bestaad nu uit een naam maar bestaad in de huidige [componenten catalogus api](https://componentencatalogus.commonground.nl/api/products/) uit veel meer details.

```json
    {
        "id": 3,
        "name": "API Testvoorziening",
        "shortDescription": "Platform om consumers enproviders van API’s te testen",
        "description": "Platform om consumers en\r\nproviders van API’s te testen",
        "detailPageImageUrl": "https://d5622d88f45b4d1fb8a6bc6d33bce28b.objectstore.eu/public/api-test.png",
        "documentationUrl": "https://github.com/VNG-Realisatie/api-test-platform",
        "demoUrl": "https://api-test.nl/",
        "bpmnProcessUrl": "",
        "isPublished": true,
        "owner": null,
        "components": []
    }
```

Een aantal van deze properties zullen we moeten overnemen in het application object.

Op dit moment zijn zowel publiccode als de componentencatalogus best ambigue met betrekking tot ownerschip en organisaties.  De componenten catalogus kent owners als conctactPersoon objecten, publiccode doet voor hetzelfde maar hanteerd voor maintanance dan weer contractor objecten.  Hier moeten we even over nadenken het voelt logisch om het te hebben over organisaties die componenten maken, ontwikkelen, beheren en gebruiken.  Maar wat doe we dan met personen die componenten ontwikkelen? Out of scope verklaren?

**Comment by lencodes** [comment]
Would be best to get the `"contact"` value as a regular string instead of markdown syntax.

So instead of: 

`
"contact": "[ruben@conduction.nl](mailto:ruben@conduction.nl)",
`

=> 

`
"contact": "ruben@conduction.nl",
`

**Reply by rubenvdlinde** [reply]
This is a commonground component catalouge thing do, in public code this is a object

**Comment by RvdKlip** [comment]
Ik weet niet goed waar bovenstaande technische verhandeling over gaat...

Onze productowner heeft een eerste aanzet aan attributen al uitgewerkt. Dit per laag van het 5-laagse model. Dus als het gaat om 'welke data willen we metadateren' moeten we denk ik dicht blijven bij de componenten die zich in die lagen bevinden. We hebben toen ook gekeken naar de huidige componenten catalogus, dus voorstel is om éérst uit te gaan van de set van Ronald.

**Reply by rubenvdlinde** [reply]
@RonaldvCortenberghe waar kunnen we jouw set vinden?

**Comment by rubenvdlinde** [comment, marked as answer]
Oke ik heb een mapping gemaakt van de set van @RonaldvCortenberghe  naar de [PublicCode ](https://conduction.stoplight.io/docs/publiccode/58v2ropv82qs8-component)Metadata en die her en der al wat uitgebreid

Op dit moment kan ik twee dingen niet zo goed kwijt

1.  Wat is het thema/domein/onderwerp (is dat iets aparts dat we moeten opnemen,  of zit het al in PublicCode)
2. Procesarchitectuur (wat willen we hiermee? is dat de procesplaat?)

| Wat                                                                     | Laag 1 (Datasets) | Laag 1 (informatiemodellen) | Laag 2 | Laag4 | Publiccode (metadata) | Property                               |
|-------------------------------------------------------------------------|-------------------|-----------------------------|--------|-------|-----------------------|----------------------------------------|
| UUID                                                                    | Ja                | Ja                          | Ja     | Ja    | Ja                    | Id                                     |
| Wie Is De Eigenaar                                                      | Ja                | Ja                          | Ja     | Ja    | Ja                    | Legal                                  |
| Wat Is De Versie                                                        | Ja                | Ja                          | Ja     | Ja    | Ja                    | SoftwareVersion                        |
| Wat Is Het Doel                                                         | Ja                | Ja                          | Ja     | Ja    | Ja                    | Description.nl                         |
| Wat Is Het Formaat (Excell, API, Csv, ….)                               | Ja                |                             |        |       | Ja/ Uitbreiding       | OutputTypes En Type                    |
| Wat Is Bovenliggend Informatiemodel                                     | Ja                |                             |        |       | Uitbreiding           | ??                                     |
| Locatie (URL)                                                           | Ja                |                             |        |       | Ja                    | LandingURL                             |
| Specificatie                                                            | Ja                |                             | Ja     |       | Ja                    | Description.[lang].apiDocumentation    |
| Afnemers?                                                               | Ja                |                             |        |       | Ja                    | UsedBy                                 |
| Relatie (met Andere Lagen Of Applicaties, Liefst Visueel Gepresenteerd) | Ja                |                             |        |       | Ja                    | DependsOn                              |
| Ondersteuning?                                                          | Ja                | Ja                          | Ja     | Ja    | Ja                    | Maintenance                            |
| Lifecycleinfo?                                                          | Ja                | Ja                          | Ja     | Ja    | Ja                    | DevelopmentStatus                      |
| Is Het Landelijk Of Rotterdams                                          |                   | Ja                          | Ja     |       | Uitbreiding           | Nl.commonground.intendedOrganisations  |
| Wat Is Het Thema/domein/onderwerp                                       |                   | Ja                          | Ja     |       | **Nee**                   | Opvangen In Gemma Extensie?            |
| Welke Entiteiten                                                        |                   | Ja                          |        |       | Ja                    | DependsOn                              |
| Soort Data (open/gemeentelijk/closed)                                   |                   | Ja                          |        | Ja    | Ja                    | Opvangen In Gemma Extensie?            |
| Soort Model?                                                            |                   | Ja                          |        |       | Ja                    | Nl.gemma.modeltype                     |
| Semantisch (UML)                                                        |                   | Ja                          |        |       | Ja                    | Nl.gemma.modeltype                     |
| Conceptueel (UML)                                                       |                   | Ja                          |        |       | Ja                    | Nl.gemma.modeltype                     |
| Logisch (UML)                                                           |                   | Ja                          |        |       | Ja                    | Nl.gemma.modeltype                     |
| Fysiek (ERD)                                                            |                   | Ja                          |        |       | Ja                    | Nl.gemma.modeltype                     |
| Applicatiefuncties (Gemma)                                              |                   |                             | Ja     | Ja    | Uitbreiding           | Nl.gemma.applicatiefunctie             |
| Status                                                                  |                   |                             | Ja     |       | Ja                    | DevelopmentStatus                      |
| Welke Gegevensmodel (laag1)                                             |                   |                             | Ja     |       | Ja                    | DependsOn                              |
| Url Broncode                                                            |                   |                             | Ja     |       | Ja                    | Url                                    |
| Output                                                                  |                   |                             | Ja     |       | Ja                    | OutputTyoes                            |
| Opleverdatum                                                            |                   |                             | Ja     | Ja    | Ja                    | ReleaseDate                            |
| End-Of-Support Datum                                                    |                   |                             | Ja     | Ja    | Ja                    | Maintenance.contractor.until           |
| Aard Proces (primair, Sturend, Ondersteunend Of Generiek)               |                   |                             |        | Ja    | Uitbreiding           | Nl.gemma.bedrijdsservices              |
| Type (bedrijfsproces, Deelproces Of Call-Activity                       |                   |                             |        | Ja    | Ja                    | Nl.gemma.bedrijfsfuncties              |
| Modeleur/opsteller (bouwer)                                             |                   |                             | Ja     | Ja    | Ja                    | Legal.authorsFile                      |
| Procesarchitectuur                                                      |                   |                             |        | Ja    | **??**                    | Wat Word Hier Bedoeld?                 |
| Relatie Met Laag 2                                                      |                   |                             |        | Ja    | Ja                    | DependsOn                              |

**Reply by matthiasoliveiro** [reply]
Vragen:
1. Wat is het thema/domein/onderwerp (is dat iets aparts dat we moeten opnemen, of zit het al in PublicCode)
Opvangen In Gemma Extensie --> akkoord
(Array)

2. Procesarchitectuur (wat willen we hiermee? is dat de procesplaat?)
Opvangen In Gemma Extensie --> akkoord

**Besloten** bovenstaande tabel als basis te gebruiken voor metadatering.

**Reply by RvdKlip** [reply]
Laag 3 en 5 nog toevoegen aan de tabel.

Daarbij geldt voor laag 5 dat 'thema/doel/onderwerp' ingevuld wordt door Gemma 'bedrijfsfunctie' en 'bedrijfsservice'
